velox/type/Type.h

```diff
@@ -658,34 +634,75 @@ class DecimalType : public ScalarType<KIND> {
     return scale_;
   }
 
+  const char* kindName() const override {
+    return "DECIMAL";
```

Comment:
Shouldn't this be `return TypeTraits<KIND>::name;`, e.g. HUGEINT or BIGINT?

Reply:
I think the function name should be changed from`kindName` -> `name`
I need this to satisfy the following check-in the SignatureBinder. Without this, a decimal function can be bound to `bigint` arguments.
```
if (typeName != actualType->kindName()) {
    return false;
  }
```

Reply:
Or we will need another API to get the logical name.

Reply:
In #4406, I introduced Type::name() for logical type name. Type::kindName() would continue to represent the physical type name.

Reply:
@majetideepak I extracted these changes into a separate PR https://github.com/facebookincubator/velox/pull/4519